numba_dppy/tests/test_dpnp_functions.py

```diff
@@ -72,6 +72,36 @@ def check_for_different_datatypes(
     return True
 
 
+def check_for_different_datatypes_array_creations(
+    fn, test_fn, dims, tys, np_all=False, matrix=None, fill_value=0, func_like=True
+):
+    for ty in tys:
+        if matrix and matrix[0]:
+            a = np.array(np.random.random(dims[0] * dims[1]), dtype=ty).reshape(
+                dims[0], dims[1]
+            )
+        else:
+            a = np.array(np.random.random(dims[0]), dtype=ty)
+        b_dppy = np.array([fill_value], dtype=ty)
+        b_numpy = b_dppy[0]
+
+        with dpctl.device_context("opencl:gpu"):
+            c = fn(a, b_dppy)
```

Comment:
This tests will need to reflect the changes suggested for `ones_like()` and `zeros_like()`